Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,242 @@
+Eg:
+```
+stringData:
+  endpoint: |-
+    ACCESS_KEY_ID: NON-BASE64-STRING
+    SECRET_ACCESS_KEY: NON-BASE64-STRING
+```
+
+##### Generated ConfigMap (sample for rook-ceph provider)
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ceph-bucket
+  namespace: rook-ceph
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  ownerReferences:
+  - name: rook-ceph
+    ...
+data:
+  BUCKET_HOST: http://my-store [1]
+  BUCKET_PORT: 80 [2]
+  BUCKET_NAME: photo-booth-62PrQ [3]
+  BUCKET_REGION: us-west-1
+  ... [4]
+```
+1. host URL.
+1. host port.
+1. unique bucket name.
+1. the above data keys are defined by the library.
+Provisioners are able to cause the lib to create additional data keys by returning the `AdditionalConfigData` field.
+
+##### App Pod (independent of provisioner)
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app-pod
+  namespace: dev-user
+spec:
+  containers:
+  - name: mycontainer
+    image: redis
+    envFrom: [1]
+    - configMapRef:
+        name: ceph-bucket [2]
+    - secretRef:
+        name: ceph-bucket [3]
+```
+1. use `env:` if mapping of the defined key names to the env var names used by the app is needed.
+1. makes available to the pod as env variables: `BUCKET_HOST`, `BUCKET_PORT`, `BUCKET_NAME`
+1. makes available to the pod as env variables: `ACCESS_KEY_ID`, `SECRET_ACCESS_KEY`
+
+##### Generated OB Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucket
+metadata:
+  name: obc-default-ceph-bucket
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+spec:
+  objectBucketSource: [1]
+    provider: ceph.rook.io/object
+  storageClassName: rook-ceph-bucket
+  claimRef: objectreference [2]
+  reclaimPolicy: {"Delete", "Retain"} [4]
+status:
+  phase: {"Pending", "Bound", "Released", "Failed"}
+```
+1. objectBucketSource is a struct containing metadata of the object store provider.
+1. objectReference to the associated OBC.
+1. reclaim policy from the Storge Class referenced in the OBC.
+
+
+##### StorageClass
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: rook-ceph-bucket
+  labels:
+    aws-s3/object [1]
+provisioner: ceph.rook.io/bucket [2]
+parameters: [3]
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-west-1
+  bucketName: ceph-bucket [4]
+reclaimPolicy: Delete [5]
+```
+1. `label`(optional) here associates this `StorageClass` to a specific provisioner.
+1. `provisioner` responsible for handling `OBCs` referencing this `StorageClass`.
+1. **all** `parameter` keys and values are specific to a provisioner, are optional, and are not validated by the `StorageClass` API.
+Fields to consider are `object-store endpoint`, `version`, etc.
+1. `bucketName` is required for access to existing buckets.
+Unlike greenfield provisioning, the brownfield bucket name appears in the `StorageClass`, not the `OBC`.
```

Comment:
Replace with:
_Change to:
_1. `bucketName` is required for access to existing buckets but is omitted when provisioning new buckets.__